Jenkinsfile-insertions

```diff
@@ -0,0 +1,88 @@
+import groovy.json.JsonSlurper
+
+pipeline{
+	agent any
+
+	stages{
+		stage('Check upstream builds succeeded'){
+			steps{
+				script{
+					def addLinksDownloadStatusUrl = httpRequest authentication: 'jenkinsKey', url: "${env.JENKINS_JOB_URL}/job/${env.RELEASE_NUMBER}/job/AddLinks-Download/lastBuild/api/json"
+					def addLinksDownloadJson = new JsonSlurper().parseText(addLinksDownloadUrl.getContent())
+					if(addLinksDownloadJson['result'] != "SUCCESS"){
+						error("Most recent AddLinks-Download build status: " + addLinksDownloadJson['result'] + ". Please complete a successful build.")
+					}
+			    	}
+				script{
+					def orthoinferenceDownloadStatusUrl = httpRequest authentication: 'jenkinsKey', url: "${env.JENKINS_JOB_URL}/job/${env.RELEASE_NUMBER}/job/Orthoinference/lastBuild/api/json"
+					def orthoinferenceJson = new JsonSlurper().parseText(orthoinferenceUrl.getContent())
+					if(orthoinferenceJson['result'] != "SUCCESS"){
+						error("Most recent Orthoinference build status: " + addLinksDownloadJson['result'] + ". Please complete a successful build.")
+					}
+			    	}
+		    	}
+	    	}
+		stage('Setup: Backup DB'){
+			steps{
+				script{
+					dir('add-links'){
+						withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
+							sh "mysqldump -u$user -p$pass ${env.RELEASE_CURRENT} > ${env.RELEASE_CURRENT}_${env.RELEASE_NUMBER}_before_addlinks.dump"
+							sh "gzip -f ${env.RELEASE_CURRENT}_${env.RELEASE_NUMBER}_before_addlinks.dump"
+						}
+					}
+				}
+			}
+		}
+		stage('Setup: Build jar file'){
+            		steps{
+		        	script{
+                    			dir('add-links'){
+						sh "mvn clean package -DskipTests"
+                    			}
+          	    		}
+            		}
+        	}
+		stage('Main: AddLinks Insertions'){
+			steps{
+				script{
+					dir('add-links'){
+						withCredentials([file(credentialsId: 'Config', variable: 'FILE')]){
+							sh "cp $FILE src/main/resources/auth.properties"
+							sh "ln -sf src/main/resources resources"
+							sh "rm -f src/main/resources/db.properties"
+					    		sh """java -cp "\$(pwd)/resources" -Dconfig.location=\$(pwd)/resources/addlinks.properties -Dlog4j.configurationFile=\$(pwd)/resources/log4j2.xml -Xmx${env.JAVA_MEM_MAX}m -jar target/AddLinks-${env.ADDLINKS_VERSION}.jar file://\$(pwd)/resources/create-references-context.xml"""
+							sh "rm -f src/main/resources/auth.properties"
+						}
+					}
+				}
+			}
+		}
+		stage('Post: Backup DB'){
+			steps{
+				script{
+					dir('add-links'){
+						withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
+							sh "mysqldump -u$user -p$pass ${env.RELEASE_CURRENT} > ${env.RELEASE_CURRENT}_${env.RELEASE_NUMBER}_after_addlinks.dump"
```

Comment:
I remember we talked about it, but I forget what our conclusion was on giving `mysql` and `mysqldump` a password outside of the command-line?  Was it necessary/doable?

Reply:
I am sure its doable, but this is probably more easily tackled when we actually deploy this on the release server, in case there are any differences in how the Jenkins user is utilized there than on my computer. For now, Jenkins is very good for making sure credentialed values show up as asterisks. Definitely can consider the use of the `my.cnf` file once its in place, though I don't know if having to switch between the local release server and curator servers are going to make things difficult, a la the `pass2production.sh` issue we had for release 70

Reply:
Sounds good - we can work on it in a new PR, if necessary, when deploying to the release server.